### PR TITLE
Update Pact pin to 4.7 release

### PR DESCRIPTION
**Pull request by jwiegley**

### CODE REVIEW

**Comment by DevopsGoth** [comment]
Failing integration tests https://github.com/kadena-io/integration-tests/actions/runs/4941771838/jobs/8834718322

File: cabal.project.freeze [hunk 1]
```diff
@@ -212,7 +212,7 @@ constraints: any.Boolean ==0.2.4,
              any.old-time ==1.1.0.3,
              any.optparse-applicative ==0.17.0.0,
              optparse-applicative +process,
-             any.pact ==4.6.0,
+             any.pact ==4.7.0,
```
Comment (emilypi) [inline cabal.project.freeze:215]: @larskuhtz are we good to modify the freeze file?

Comment (edmundnoble) [reply cabal.project.freeze:215]: I'll ask Louis.

Comment (edmundnoble) [reply cabal.project.freeze:215]: Yes we are, we haven't checked in the release freeze file yet.

**Review by edmundnoble — APPROVED** [review]

**Comment by edmundnoble** [comment]
The RC has been keeping up with Pact master before 4.7 was released, all we should need is integration tests working.

**Comment by edmundnoble** [comment]
Integration tests work as of Hee Kyun's latest commit. Merging.

**Pull request merged**